great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -218,10 +234,19 @@ def get_engine(self) -> sqlalchemy.Engine:
                 connection_string: str | dict = model_dict.pop("connection_string")
 
                 if isinstance(connection_string, str):
-                    self._engine = sa.create_engine(connection_string, **kwargs)
+                    url = sa.engine.url.make_url(connection_string)
+                    url = url.update_query_dict(
+                        query_parameters={"application": snowflake_partner_application}
+                    )
+                    self._engine = sa.create_engine(
+                        url,
+                        **kwargs,
+                    )
                 else:
                     self._engine = self._build_engine_with_connect_args(
-                        **connection_string
+                        application=snowflake_partner_application,
+                        **connection_string,
+                        **kwargs,
```

Comment:
@Kilo59 - I noticed **kwargs was missing here. Assumed we still wanted it since we're still passing **kwargs to engine for str connection_string, and we were passing **kwargs here before. Is this correct?

Reply:
I think the kwargs are passed elsewhere.

Reply:
I thought they might be too, but they end up getting filtered out in `_get_connect_args`, since `kwargs` is defined in parent class:

```
    def _get_connect_args(self) -> dict[str, str | bool]:
        excluded_fields: set[str] = set(SQLDatasource.__fields__.keys())
        # dump as json dict to force serialization of things like AnyUrl
        return self._json_dict(exclude=excluded_fields, exclude_none=True)
```